pandas/core/ops.py

```diff
@@ -515,7 +515,8 @@ def _convert_to_array(self, values, name=None, other=None):
 
             # a datelike
             elif isinstance(values, pd.DatetimeIndex):
-                values = values.to_series()
+                # TODO: why are we casting to_series in the first place?
```

Comment:
and if you change this does it work?

Reply:
Tentative yes, but only with the change made by `_get_series_result_name` below.

Reply:
Update: tinkering with this _and_ doing the threading requested below don't play nicely together.  We should keep this as-is and address separately.  This fixes a bug and we should call it a win.  There are a bunch of these bug-fixes to get in before we can clean up the mess that is _TimeOp.